File: src/types/charts.ts
export interface Transaction {
  id: number;
  account_id: number;
  account_name: string;
  type: string;
  quantity: number;
  unit_price: number;
  description: string;
  date: Date;
}

export interface Chart {
  description: string;
  quantity: number;
  rate: number;
  growth: number;
  growth_percentage: number;
  items: Transaction[];
  next: boolean;
}

export enum PeriodEnum {
  Week = '1w',
  Month = '1m',
  Year = '1y',
  All = 'All',
}

export interface FetchChartProps {
  location_id: number | null;
  account_id: number | null;
  period: PeriodEnum;
}

File: src-tauri/src/models/charts.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

pub enum PeriodEnum {
    Day,
    Week,
    Month,
    Year,
}

impl PeriodEnum {
    pub fn as_str(&self) -> &str {
        match self {
            PeriodEnum::Day => "1w",
            PeriodEnum::Week => "1m",
            PeriodEnum::Month => "1y",
            PeriodEnum::Year => "All",
        }
    }
}

pub struct ChartFilters {
    pub location_id: Option<i32>,
    pub account_id: Option<i32>,
    pub period: PeriodEnum,
}
